### PR TITLE
test: coverage for rolling average time struct

### PR DESCRIPTION
**Pull request by jorgeantonio21**
Description

- Add test coverage at `base_layer/core/src/common/...`

Motivation and Context

- Some additional coverage needs to be added to the code base . This PR adds a few of additional tests to cover logic regarding RollingAverageTime struct, etc.

How Has This Been Tested?

- We added tests for each of the present methods in the modified files.

### CODE REVIEW

File: base_layer/core/src/common/byte_counter.rs [hunk 1]
```diff
@@ -48,3 +48,24 @@ impl io::Write for ByteCounter {
...
+        let mut byte_counter = ByteCounter::new();
+        let buf = [0u8, 1u8, 2u8, 3u8];
+        let new_count = byte_counter.write(&buf).unwrap();
+        assert_eq!(byte_counter.get(), new_count);
+    }
+
+    #[test]
+    fn flush_test() {
+        let mut byte_counter = ByteCounter::new();
+        let flushed = byte_counter.flush().unwrap();
```
Comment (stringhandler) [inline base_layer/core/src/common/byte_counter.rs]: I think add some bytes to it as well

```suggestion
        let flushed = byte_counter.flush().unwrap();
```

**Review by stringhandler — DISMISSED** [review]
Nice, just need the CI to pass

File: base_layer/core/src/common/byte_counter.rs (hunk 1 above)
Comment (SWvheerden) [reply base_layer/core/src/common/byte_counter.rs]: Do something like this:
```
        let buf = [0u8, 1u8, 2u8, 3u8];
        let new_count = byte_counter.write(&buf).unwrap();
```

**Review by SWvheerden — COMMENTED** [review]
This looks good, just add those bytes. 
Please run `cargo fmt` as well and push that up.

File: base_layer/core/src/common/byte_counter.rs (hunk 1 above)
Comment (jorgeantonio21) [reply base_layer/core/src/common/byte_counter.rs]: Hi guys, thank you for the comments. I did add some bytes in the `flush_test` function, could you guys look it up and see if it is okay.

**Comment by jorgeantonio21** [comment]
I did run the `cargo fmt` and applied the comments above

**Review by SWvheerden — APPROVED** [review]
LGTM

**Review by stringhandler — APPROVED** [review]

**Pull request merged**